hazelcast-sql/src/main/java/com/hazelcast/jet/sql/impl/PlanExecutor.java

```diff
@@ -187,6 +188,14 @@ SqlResult execute(CreateDataConnectionPlan plan) {
         if (dlService.existsConfigDataConnection(plan.name())) {
             throw new HazelcastException("Cannot replace a data connection created from configuration");
         }
+
+        try {
+            // checks if type is correct
+            dlService.classForDataConnectionType(plan.type());
+        } catch (Exception e) {
+            sneakyThrow(e);
```

Comment:
classForDataConnectionType does not declare exceptions, sneakyThrow is redundant